src/session.ps1

```diff
@@ -160,3 +161,42 @@ function Disconnect-XoSession {
     $script:XoRestParameters = $null
 }
 New-Alias -Name Disconnect-XenOrchestra -Value Disconnect-XoSession
+
+function Set-XoDefaultLimit {
+    <#
+    .SYNOPSIS
+        Set the default limit for XO query cmdlets.
+    .DESCRIPTION
+        Sets the default limit for all Get-Xo* cmdlets that support a -Limit parameter.
+        This setting persists for the current PowerShell session.
+    .PARAMETER Limit
+        The default limit to use. Set to 0 for unlimited results.
+    .EXAMPLE
+        Set-XoDefaultLimit -Limit 50
+        Sets the default limit to 50 items for all query cmdlets.
+    .EXAMPLE
+        Set-XoDefaultLimit -Limit 0
+        Sets cmdlets to return all items by default.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [int]$Limit
+    )
+
+    # Initialize the variable if it doesn't exist
+    if ($null -eq $script:XO_DEFAULT_LIMIT) {
+        $script:XO_DEFAULT_LIMIT = 25
+        Write-Verbose "Default limit was not initialized, setting initial value to 25"
+    }
+
+    # Set the new limit
+    $oldLimit = $script:XO_DEFAULT_LIMIT
+    $script:XO_DEFAULT_LIMIT = $Limit
+    
```

Comment:
`$script:XO_DEFAULT_LIMIT` should be a constant that's not touched by user commands. The default session limit should be named e.g. `$script:XoLimit` or `$script:XoSessionLimit`. It should also be reset on disconnection.